.github/workflows/modules_generator.yml

```diff
@@ -0,0 +1,99 @@
+---
+name: modules generator
+
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.sha }}'
+  cancel-in-progress: true
+
+on:
+  # push:
+  #   branches:
+  #     - main
+  #     - 'stable-*'
+  # pull_request_target:
+  #   types:
+  #     - labeled
+  #     - unlabeled
+  #     - opened
+  #     - reopened
+  #     - synchronize
+  pull_request:
+
+jobs:
+  generate:
+    runs-on: ubuntu-latest
+    env:
+      python_version: '3.9'
+      ansible_version: 'stable-2.11'
+      collection_path: "amazon_cloud"
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.head_ref }}
+          path: ${{ env.collection_path }}
+
+      - name: Set up Python ${{ env.python_version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.python_version }}
+
+      - name: Install python modules
+        run: >-
+          python3 -m pip install
+          boto3
+          botocore
+          black==22.3.0
+          jsonschema
+          jinja2==3.0.3
+          q
+          https://github.com/ansible/ansible/archive/${{ env.ansible_version }}.tar.gz --disable-pip-version-check
+        shell: bash
+
+      - name: Install required ansible collections
+        run: |
+          ansible-galaxy collection install git+https://github.com/ansible-community/ansible.content_builder.git
+          ansible-galaxy collection install ansible.utils
+        shell: bash
+
+      - name: Generate playbook to execute
+        run: |
+          echo "---" > build.yaml
+          echo "- hosts: localhost"  >> build.yaml
+          echo "  gather_facts: yes" >> build.yaml
+          echo "  roles:"            >> build.yaml
+          echo "  - ansible.content_builder.run" >> build.yaml
+        shell: bash
+
+      - name: Generate manifest file
+        run: |
+          echo "---" > manifest.yaml
```

Comment:
Not sure though, but it's probably better to host a manifest.yml file inside the collection where module_version is automatically updated in the case of a new release (as we do for the agent string). The collection contains the api schemas already in the config folder and we should probably put the manifest file there is this makes sense to you. What do you think @abikouo @jillr @gravesm?

Reply:
I believe that if something happens with the automated release process, we can manually do a release since the collection will host all the necessary files (we won't need anymore to create a manifest on our own).